tiledb/cf/engines/netcdf4_engine.py

```diff
@@ -632,48 +633,56 @@ def _from_group_to_attr_per_array(
         """
         converter = cls(default_input_file, default_group_path)
+        coord_names = set()
+        tiles_by_var = {} if tiles_by_var is None else tiles_by_var
+        tiles_by_dims = {} if tiles_by_dims is None else tiles_by_dims
         if coords_to_dims:
-            raise NotImplementedError
+            for ncvar in netcdf_group.variables.values():
+                if ncvar.ndim == 1 and ncvar.dimensions[0] == ncvar.name:
+                    converter.add_coord_to_dim_converter(ncvar)
+                    coord_names.add(ncvar.name)
         for ncvar in netcdf_group.variables.values():
-            for dim in ncvar.get_dims():
-                if dim.name not in converter.dim_names:
-                    converter.add_dim_to_dim_converter(
-                        dim,
-                        unlimited_dim_size,
-                        dim_dtype,
-                    )
+            if ncvar.name in coord_names:
+                continue
             if not ncvar.dimensions:
-                array_name = (
-                    "scalars" if "scalars" not in netcdf_group.variables else "_scalars"
-                )
-                if array_name not in converter.array_names:
+                if scalar_array_name in netcdf_group.variables:
+                    raise ValueError(
+                        f"Cannot name array of scalar values `{scalar_array_name}`. An"
+                        f" array with that name already exists."
+                    )
+                if scalar_array_name not in converter.array_names:
                     converter.add_scalar_dim_converter("__scalars", dim_dtype)
-                    converter.add_array("scalars", ("__scalars",))
+                    converter.add_array(scalar_array_name, ("__scalars",))
+                converter.add_var_to_attr_converter(ncvar, scalar_array_name)
             else:
-                if tiles_by_var is not None and ncvar.name in tiles_by_var:
-                    array_tiles = tiles_by_var[ncvar.name]
-                elif tiles_by_dims is not None and ncvar.dimensions in tiles_by_dims:
-                    array_tiles = tiles_by_dims[ncvar.dimensions]
-                else:
-                    chunks = ncvar.chunking()
-                    array_tiles = (
-                        None
-                        if chunks is None or chunks == "contiguous"
-                        else tuple(chunks)
-                    )
+                for dim in ncvar.get_dims():
+                    if dim.name not in converter.dim_names:
+                        converter.add_dim_to_dim_converter(
+                            dim,
+                            unlimited_dim_size,
+                            dim_dtype,
+                        )
+                array_tiles = tiles_by_var.get(
+                    ncvar.name,
+                    tiles_by_dims.get(ncvar.dimensions, get_variable_chunks(ncvar)),
+                )
```

Comment:
Can `tiles_by_var`/`tiles_by_dims` return a falsy value (`None, (), 0`)? If not I'd rewrite with `or`:

```py
array_tiles = (
    tiles_by_var.get(ncvar.name) or 
    tiles_by_dims.get(ncvar.dimensions) or 
    get_variable_chunks(ncvar)
)

Reply:
They can return a `None` value.